tests/background/tab-state-test.js

```diff
@@ -136,13 +137,15 @@ describe('TabState', () => {
     let getStub;
     const INITIAL_WAIT_MS = 1000;
     const MAX_WAIT_MS = 3000;
+    const EXPIRATION_CACHE_MS = 3000;
 
     beforeEach(() => {
       clock = sinon.useFakeTimers();
       getStub = sinon.stub();
       $imports.$mock({
         './uri-info': {
-          getAnnotationCount: getStub,
+          fetchAnnotationCount: getStub,
+          shouldQueryUri,
```

Comment:
Rather than calling through to `shouldQueryUri` directly here, you could mock it to directly throw a `BadgeUrlError`. This avoids coupling this test to the details of `shouldQueryUri`.

Reply:
Good idea!